### DISCUSSION TITLE
'factor' on 64-bit machines

### DISCUSSION THREAD

**Discussion by nerdhardest** [uutils/coreutils#4713] [General]
I tested this Rust coreutils version of 'factor', and found that it
was limited to numbers ~2^64.

According to Gnu, "If the number to be factored is small (less than 2^{127} on typical machines), factor uses a faster algorithm" and
"factoring the eighth Fermat number 2^{256} + 1 takes about 14 seconds".

https://www.gnu.org/software/coreutils/manual/html_node/factor-invocation.html

Also, this website below indicates that 'factor' on 64-bit machines should be capable of at least ~2^127 :

https://wiki.linuxquestions.org/wiki/Factor

---
I realize that this is a relatively minor nit, but Rust has excellent
bignum arithmetic capabilities, so increasing the precision of
'factor' shouldn't be a big problem for Rust coreutils.

**Comment by tertsdiepraam** [comment]
You're absolutely right. There's an open issue for this: https://github.com/uutils/coreutils/issues/1559
